engine/runtime/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -435,9 +436,8 @@ class Compiler(
           val shouldStoreCache =
             irCachingEnabled && !context.wasLoadedFromCache(module)
           if (
-            shouldStoreCache && !hasErrors(module) && !context.isInteractive(
-              module
-            )
+            shouldStoreCache && !hasErrors(module) &&
+            !context.isInteractive(module) && !context.isSynthetic(module)
```

Comment:
Why is this change needed? Why skip serialization of synthetic modules? Isn't it possible that this leads to some inconsistencies? Could synthetic modules be generated on the fly during deserialization of other modules?

Reply:
> Why skip serialization of synthetic modules? Isn't it possible that this leads to some inconsistencies?

I don't remember the exact reasons but I guess having a mixed scenario of cached and uncached IRs might be one.

Reply:
According to @hubertp, synthetic modules aren't supposed to be serialized (moreover they cannot be serialized, as they yield exceptions during serialization).